### PR TITLE
Integration tests for getTransactionsByAddress and getUTXO

### PR DESCRIPTION
**Pull request by abvgedeika**
Issue being fixed or implemented
add integration tests for getTransactionsByAddress and getUTXO

What was done
new integration tests have been added 

What has been tested
Nothing

What needs to be tested
Integration tests for  getTransactionsByAddress and getUTXO passed

### CODE REVIEW

File: test/integration/e2e/index.js [hunk 1]
```diff
@@ -277,7 +273,7 @@ describe('basic E2E tests', () => {
                 .setPubKeyIdFromPrivateKey(alicePrivateKey).sign(alicePrivateKey);
 
             let inputs = await dapiClient.getUTXO(faucetAddress);
-            expect(inputs.items).to.have.lengthOf(2);
+            expect(inputs.items).to.have.lengthOf(1);
 
             const transaction = Transaction()
                 .setType(Transaction.TYPES.TRANSACTION_SUBTX_REGISTER)
```
Comment (Cofresi) [inline test/integration/e2e/index.js]: I'm wondering why inputs.items should always have just 2 elements? Doesn't this depend on how many txs have been made to the faucetAddress and could be anything between 0 and 999?

Comment (abvgedeika) [reply test/integration/e2e/index.js]: good question. indeed in these tests size of inputs.items equals 1 and doesn't make sense to use the latest input as: inputs.items.slice(-1)[0]. I'll add separate tests that will use inputs as array with size >1 and < 999

Comment (abvgedeika) [reply test/integration/e2e/index.js]: @Cofresi I've added new tests to verify how many inputs we could pass to Transaction https://github.com/dashevo/dapi-client/pull/53/commits/46eb4e501e91bd29b4754698b976188de63f50a3

1) with last 15 inputs( inputs.items.slice(-15)) transaction is rejected with 'rate limited free transaction'
2) with last 15 inputs everything is fine

Do you think this is a valid behavior?

File: test/integration/e2e/index.js [hunk 2]
```diff
@@ -49,6 +49,7 @@ describe('basic E2E tests', () => {
     let aliceRegTxId;
 
     let aliceUser;
+    let bobContactRequest;
```
Comment (shumkov) [inline test/integration/e2e/index.js]: Why do we duplicate here our devnet e2e tests?

File: test/integration/features/index.js [hunk 3]
```diff
@@ -1082,13 +1082,23 @@ describe('features', () => {
         bobRegTxId = result;
 
         bobPreviousST = result;
+
+        //now we verify that getUTXO.items not empty when no new block generated
+        const utxo = await dapiClient.getUTXO(faucetAddress);
+        console.log(utxo);
```
Comment (shumkov) [inline test/integration/features/index.js]: Please remove console logs

File: test/integration/e2e/index.js (hunk 2 above)
Comment (abvgedeika) [reply test/integration/e2e/index.js]: we need tests for verification dapi and others components on a regular basis, ideally is on every commit. we don't need to verify functional tests only after deployment in devnet.

File: test/integration/features/index.js (hunk 3 above)
Comment (abvgedeika) [reply test/integration/features/index.js]: I log only this case to show value of getUtxo for existing bug to track this value

Comment (shumkov) [reply test/integration/features/index.js]: But if it's only for debug purposes it should not go to the master then.

File: test/integration/e2e/index.js (hunk 1 above)
Comment (Cofresi) [reply test/integration/e2e/index.js]: I think there is something wrong with this. Still investigating, but this is where its coming from:
https://github.com/dashpay/dash/blob/master/src/validation.cpp#L841
It means with 15 inputs the size of the transaction needs a bigger fee to pass this validation test. So I guess what decides that the test errors with this message is not only the mere number of 15 inputs, but also how much decide to pay the miners. In this specific tx, even if you add a lot of fees as credits to the OP_RETURN output of the TRANSACTION_SUBTX_REGISTER, this still doesn't count to the fees you are paying the miner. The classical tx fees are calculated as 
cf: https://github.com/dashpay/dips/blob/master/dip-0005.md#subscription-transaction-fees
My hypothesis right now is that adding the credit output doesn't actually count as an output sum and that all of the available input amount is added to the change output which would leave nothing to the miner.
Will investigate further...

File: test/integration/features/index.js [hunk 4]
```diff
@@ -539,4 +545,718 @@ describe('retry policy', () => {
...
+        "fromHeight": fromHeight,
+        "items": [],
+        "to": 1,
+        "totalItems": 1
+      });
+    });
+
+
+      it('should throw error when call getTransactionsByAddress without params', async function it() {
+        let err = '';
```
Comment (Cofresi) [inline test/integration/features/index.js]: Formatting seems to be off in some tests.

Comment (abvgedeika) [reply test/integration/features/index.js]: in all our tests eslint formatting is off https://github.com/dashevo/dapi-client/blob/master/.eslintignore#L1

**Pull request merged**